include/boost/url/impl/params.hpp

```diff
@@ -285,6 +287,7 @@ assign(FwdIt first, FwdIt last,
 //
 //------------------------------------------------
 
+inline
```

Comment:
these were already marked inline in the first declaration, or else how do you think that this would link?

Reply:
I moved the keywords to the implementation. This was generating a warning in static analysis: https://isocpp.org/wiki/faq/inline-functions#where-to-put-inline-keyword

Reply:
hmm......... where is this warning? what does boost.json do?

Reply:
> hmm......... where is this warning? 

It's not a warning in our tooling. I don't even think we have common static analysis in our tooling. This is local. The tool points me to the isocpp guidelines in https://isocpp.org/wiki/faq/inline-functions#where-to-put-inline-keyword

The guideline seems reasonable unless there's another motivation I don't know about. But I can undo that if you want, of course. I'll just disable it locally so it stops annoying me.

Reply:
if we change it, we have to change it not only globally in url but also http_proto, json, and beast

Reply:
> if we change it, we have to change it not only globally in url but also http_proto, json, and beast

Well... that's not up to me then. I'll undo it and disable the warning locally.

Reply:
Lets open an issue for it, so we can track it

Reply:
It's already undone, but you can let me know if I should redo it at any time. 

My understanding is the isocpp guideline is correct and the reason Boost.URL does it seems to be because it's replicating a pattern from ASIO that doesn't apply here. To the best of my knowledge, Boost.URL doesn't have a macro intended to enable and disable these specifiers depending on how the ipp files are going to be used.

In any case, I don't think the issue is useful at this point because you have already stated you don't consider this to be an issue in this lib.